health-monitors/gpu-health-monitor/gpu_health_monitor/dcgm_watcher/dcgm.py

```
@@ -239,6 +258,88 @@ def _perform_health_check(self, dcgm_group: pydcgm.DcgmGroup) -> tuple[dict[str,
+        is unset, or no metadata reader is configured.
+        """
+        if not self._thermal_margin_enabled or self._field_group is None or self._metadata_reader is None:
+            return None
+
+        monitor = DCGM_FIELDS_MONITORING["gputemplimitmonitoringenabled"]
+        margin_details = types.HealthDetails(status=types.HealthStatus.PASS, entity_failures={})
+
+        try:
+            with metrics.dcgm_api_latency.labels("dcgm_field_153_get_latest").time():
+                field_values = dcgm_group.samples.GetLatest(self._field_group)
+        except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)